vignettes/design.qmd

```diff
@@ -58,4 +58,49 @@ if the processing time per file is substantial.
 ```{mermaid}
 %%| fig-alt: "A flowchart showing the data flow within `registers2parquet`, showing how SAS files are imported, potentially joined, and then exported as Parquet files with a year partitioning."
 %%| file: flow.mmd
-```
+```
+
+## Expected flow
+
+We expect the flow of using the `registers2parquet` package to convert
+register SAS files to Parquet files will be as follows:
+
+```{mermaid}
+%%| label: fig-flow
+%%| fig-cap: "Flow of the expected flow using the `convert_register_to_parquet()` package to convert register SAS files to Parquet files."
+%%| fig-alt: "A flowchart showing the expected flow of converting register SAS files to Parquet files using the `registers2parquet` package. It includes identifying paths to the relevant SAS files using the helper functions, calling the main function `convert_register_to_parquet()`, which then outputs the path to the written Parquet files."
+flowchart TD
+
+    identify_paths("Identify register path(s)<br>from within R<br>with list helper functions")
+    path[/"path<br>[Character vector]"/]
+    output_path[/"output_path<br>[Character vector]"/]
+    convert_register_to_parquet("convert_register_to_parquet()")
+    output[/"Output<br>[Character vector]<br>Path to written Parquet file"/]
+
+    %% Edges
+    identify_paths -.-> path --> convert_register_to_parquet
+    output_path --> convert_register_to_parquet
+
+    output_path --> convert_register_to_parquet
+    convert_register_to_parquet --> output
+
+    %% Styling
+    style identify_paths fill:#FFFFFF, color:#000000, stroke-dasharray: 5 5
+```
+
+We expect a flow of reading a Parquet register created by the
+`registers2parquet` package into an R session to be as follows:
+
+```{mermaid}
+%%| label: fig-flow-use
+%%| fig-cap: "Flow of the expected flow of reading a Parquet register that was created with the `registers2parquet` package."
+%%| fig-alt: "A flowchart showing the expected flow of reading a Parquet register created with the `registers2parquet` package. It includes using the register name to read a specific register into the R session using the `read_register_parquet()` function, which outputs the register as a DuckDB table."
+flowchart TD
+
+    path[/"path<br>[Character vector]"<br>Path to register dir/]
+    read_register_parquet("read_register_parquet()")

```

Comment:
```suggestion
    read_register("read_register()")

```

I don't think this is necessary to be so long. Especially if we only expose it as the only read function.